tests/test-output-flex-update.cpp

```diff
@@ -38,8 +39,51 @@ struct options_slim_expire
     }
 };
 
+struct options_slim_schema
+{
+    static options_t options()
+    {
+        auto conn = db.db().connect();
+        // Create limited user (if it doesn't exist yet),
+        // which we need to test that the public schema won't be touched.
+        // If the public schema is tried to be modified at any point, this user won't have the
+        // necessary permissions, and hence the test will fail.
+        conn.exec(R"(
+DO
+$$
+BEGIN
+   IF NOT EXISTS (SELECT FROM pg_catalog.pg_roles WHERE rolname = 'limited') THEN
+      CREATE ROLE limited LOGIN;
+   END IF;
+END
+$$;
+                  )");
+        conn.exec("REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA public FROM "
+                  "PUBLIC, limited;");
+        conn.exec("REVOKE CREATE ON SCHEMA public FROM PUBLIC, limited;");
+        conn.exec(
+            "CREATE SCHEMA IF NOT EXISTS myschema AUTHORIZATION limited;");
+        conn.close();
+        return testing::opt_t()
+            .slim()
+            .flex(conf_file)
+            .schema("myschema")
+            .user("limited");
+    }
+};
+
+// Return a string with the schema name prepended to the table name.
+std::string with_schema(char const *name, options_t const *options)
```

Comment:
We are not changing the options, so we should probably use a `const &` instead of `const *` as is done elsewhere in the osm2pgsql code. This will also make calling the function simpler.